cdap-security/src/main/java/co/cask/cdap/security/tools/GeneratedCertKeyStores.java

```diff
@@ -0,0 +1,156 @@
+import java.util.Date;
+
+/**
+ * Utility class with methods for generating a X.509 self signed certificate
+ * and creating a Java key store with a self signed certificate.
+ */
+public final class GeneratedCertKeyStores {
+  private static final String KEY_PAIR_ALGORITHM = "RSA";
+  private static final String SECURE_RANDOM_ALGORITHM = "SHA1PRNG";
+  private static final String SECURE_RANDOM_PROVIDER = "SUN";
+
+  /* This is based on https://www.ietf.org/rfc/rfc1779.txt
+     CN      CommonName
+     L       LocalityName
+     ST      StateOrProvinceName
+     O       OrganizationName
+     OU      OrganizationalUnitName
+     C       CountryName
+     STREET  StreetAddress
+
+     All fields are not required.
+    */
+  private static final String DISTINGUISHED_NAME = "CN=CDAP, L=Palo Alto, C=US";
+  private static final String SIGNATURE_ALGORITHM = "MD5withRSA";
+  private static final String SSL_KEYSTORE_TYPE = "JKS";
+  private static final String CERT_ALIAS = "cert";
+  private static final int KEY_SIZE = 2048;
+  private static final int VALIDITY = 999;
+
+  /* private constructor */
+  private GeneratedCertKeyStores() {}
+
+  /**
+   * Create a Java key store with a stored self-signed certificate.
+   * @return Java keystore which has a self signed X.509 certificate
+   */
+  public static KeyStore getSSLKeyStore(SConfiguration sConf) {
+    KeyStore keyStore;
+    String password = sConf.get(Constants.Security.Ssl.SSL_KEYSTORE_PASSWORD);
+    try {
+      String keyPairAlgorithm = sConf.get(Constants.Security.Ssl.KEY_PAIR_ALGORITHM,
+                                          KEY_PAIR_ALGORITHM);
+      KeyPairGenerator keyGen = KeyPairGenerator.getInstance(keyPairAlgorithm);
+      String randomAlgorithm = sConf.get(Constants.Security.Ssl.SECURE_RANDOM_ALGORITHM,
+                                         SECURE_RANDOM_ALGORITHM);
+      String randomProvider = sConf.get(Constants.Security.Ssl.SECURE_RANDOM_PROVIDER,
+                                        SECURE_RANDOM_PROVIDER);
+      SecureRandom random = SecureRandom.getInstance(randomAlgorithm, randomProvider);
+      keyGen.initialize(KEY_SIZE, random);
+      // generate a key pair
+      KeyPair pair = keyGen.generateKeyPair();
+      int validity = sConf.getInt(Constants.Security.Ssl.CERT_VALIDITY, VALIDITY);
+      String distinguishedName = sConf.get(Constants.Security.Ssl.CERT_DISTINGUISHED_NAME, DISTINGUISHED_NAME);
+      String signatureAlgo = sConf.get(Constants.Security.Ssl.SIGNATURE_ALGORITHM, SIGNATURE_ALGORITHM);
+
+      X509Certificate cert = getCertificate(distinguishedName, pair, validity, signatureAlgo);
+
+      keyStore = KeyStore.getInstance(sConf.get(Constants.Security.Ssl.SSL_KEYSTORE_TYPE, SSL_KEYSTORE_TYPE));
+      keyStore.load(null, password.toCharArray());
+      keyStore.setKeyEntry(CERT_ALIAS, pair.getPrivate(), password.toCharArray(),
+                           new java.security.cert.Certificate[]{cert});
+    } catch (Throwable e) {
```

Comment:
It would be good to only catch `Exception` here, since `Throwable` includes `Error`s too